src/modules/expression/binop/range.rs

```diff
@@ -59,7 +60,10 @@ impl TranslateModule for Range {
         let from = self.from.translate(meta);
         let to = self.to.translate(meta);
         if self.neq {
-            let to_neq = translate_computation(meta, ArithOp::Sub, Some(to), Some("1".to_string()));
+            let to_neq = match &self.to.value.as_ref().unwrap() {
+                ExprType::Number(_) => (to.parse::<usize>().unwrap() - 1).to_string(),
```

Comment:
This part should be better handled. The `ExprType::Number` can be for instance:
- `0.1`
- `-123`

In each case this `unwrap` will panic. I think that you should first parse left hand side and right hand side to `usize` without panic and also compare if `a < b`, then return the `b-1` and otherwise fallback to the `translate_computation` method.

The problem with floating point numbers will come to end when we introduce the integer types. I'm finishing the `divide` function for fixed precision real numbers in Bash. I think that we could get rid of `bc` dependency.

Reply:
I think this is out of scope for this PR, as it's part of a wider problem.

Reply:
In the meantime, I've changed the `unwrap()` call to one of my favourite Rust features, `unwrap_or_default()`.

Reply:
Seems to work as well as can be expected with negative and float values:
```
hwalters@Ghostwheel ~/git/amber-lang (range-neq-opt) 
$ cat range.ab 
#!/usr/bin/env amber
loop num in -5..-1 { echo num }
loop num in 2.3..=5.7 { echo num }
```
```
hwalters@Ghostwheel ~/git/amber-lang (range-neq-opt) 
$ ./range.ab 
-5
-4
-3
-2
2.3
3.3
4.3
5.3
```